WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -173,7 +182,9 @@ private void unpackPostIntent(@NonNull Intent intent) {
             if (!hasPendingOrInProgressMediaUploadsForPost(post)) {
                 mPostUploadHandler.upload(post);
             } else {
-                sPostsWithPendingMedia.add(post);
+                List<MediaModel> pendingMedia = MediaUploadHandler.getPendingOrInProgressMediaUploadsForPost(post);
+                UploadingPost uploadingPost = new UploadingPost(post, pendingMedia);
```

Comment:
As of 1c759a8, keeping track of posts that were cancelled while media were pending, so that we remember the existing media list if we re-upload the post, even if some of these media have since finished uploading (while also keeping track of any new media the user added during the last edit session).